pkg/detectors/mongodb/mongodb.go

```diff
@@ -46,30 +51,47 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		}
 
 		if verify {
-			func() {
-				ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
-				defer cancel()
-				client, err := mongo.Connect(ctx, options.Client().ApplyURI(resMatch))
-				if err != nil {
-					return
-				}
-				defer func() {
-					if err := client.Disconnect(ctx); err != nil {
-						return
-					}
-				}()
```

Comment:
@ahrav i _think_ this was just discarding the result of `client.Disconnect` so i simplified it a bit when i moved it, but give a shout if the new version looks wrong

Reply:
looks good!